### PR TITLE
collectd 5.5.0

### PR DESCRIPTION
**Pull request by nijikon**

### CODE REVIEW

File: Library/Formula/collectd.rb [hunk 1]
```diff
@@ -25,6 +19,13 @@ class Collectd < Formula
     depends_on "autoconf" => :build
   end
 
+  # Will fail against Java 1.7
+  option "with-java", "Enable Java 1.6 support"
```
Comment (mikemcquaid) [inline Library/Formula/collectd.rb]: You can do this with a `depends_on :java => [:optional, "1.6"]` I believe?

File: Library/Formula/collectd.rb [hunk 2]
```diff
@@ -80,4 +81,16 @@ def plist; <<-EOS.undent
...
+  test do
+    begin
+      fork { system sbin/"collectd" }
+      sleep 2
+      assert File.exist?("#{var}/run/collectd.pid")
+    ensure
+      pid = File.read("#{var}/run/collectd.pid").to_i
+      Process.kill("SIGINT", pid)
+      Process.wait(pid)
+    end
```
Comment (mikemcquaid) [inline Library/Formula/collectd.rb]: Test is failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/30168/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_collectd/

File: Library/Formula/collectd.rb (hunk 1 above)
Comment (nijikon) [inline Library/Formula/collectd.rb]: There is already line ```depends_on :java => ["1.6", :optional]```. What do you want me to do here?

File: Library/Formula/collectd.rb [hunk 3]
```diff
@@ -80,4 +81,14 @@ def plist; <<-EOS.undent
     </plist>
     EOS
   end
+
+  test do
+    begin
+      system sbin/"collectd"
+      sleep 2
+      assert File.exist?("#{var}/run/collectd.pid")
```
Comment (xu-cheng) [inline Library/Formula/collectd.rb]: This test will fail under sandbox. As we don't allow to write files outside of temporary directory during the test. Can you set this file to builldpath by argument or something?

**Comment by nijikon** [comment]
@xu-cheng I rewrote the test, please verify.

**Comment by xu-cheng** [comment]
Thanks @nijikon

**Pull request closed without merging**